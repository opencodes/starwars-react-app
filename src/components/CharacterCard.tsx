import React from 'react'
import { Link } from 'react-router-dom'
import { useAppContext } from '../context/AppContext'
import { ICharacter } from '../interface/swapi.interface'

type Props = {
    character: ICharacter,
    onFavouriteClick: (character: ICharacter) => void
}

const CharacterCard = ({ character, onFavouriteClick }: Props) => {
    const { planets } = useAppContext();

    return (
        <div className="card"  >
            <img src={character.imgURL} className="card-img-top" alt="..." />
            <div className="card-body">
                <h5 className="card-title">{character.name}</h5>
                <div style={{ height: 50 }}>
                    <p > Gender : <span className="material-symbols-outlined">{character.gender !== 'n/a' ? character.gender : ''}</span></p>
                    <p >Home Planet : {character.homeworld}</p>
                </div>
            </div>
            <div className="card-body">
                <div className="d-flex justify-content-between">
                    <Link to={`/characters/${character.id}`} >
                        <button type="button" className="card-link btn btn-primary" >View </button></Link>
                    <button type="button" className="card-link btn btn-warning" onClick={() => onFavouriteClick(character)} >
                        <span className="material-symbols-outlined">favorite</span>
                    </button>
                </div>
            </div>
        </div>
    )
}

export default CharacterCard