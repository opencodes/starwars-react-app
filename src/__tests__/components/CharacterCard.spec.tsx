import CharacterCard from '../../components/CharacterCard'
import { mockCharacter } from '../../__mock__/mockCharacter';
import { renderWithRouter } from '../test-utils';
import { screen, fireEvent } from '@testing-library/react';

describe('Page', () => {
    test('should render Page correctly', () => {
        const props = {
            character: mockCharacter,
            onFavouriteClick: jest.fn(),
            disableFav: true
        }
        const { asFragment } = renderWithRouter(<CharacterCard {...props} />);
        expect(asFragment).toMatchSnapshot();
    });
    test('should trigger onFavouriteClick event', () => {
        const props = {
            character: mockCharacter,
            onFavouriteClick: jest.fn(),
            disableFav: true
        }
        renderWithRouter(<CharacterCard {...props} />);
        const previousBtn = screen.getByTestId('favorite-btn')
        fireEvent.click(previousBtn)
    });
});
