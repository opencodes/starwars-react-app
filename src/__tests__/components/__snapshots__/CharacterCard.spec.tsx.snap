// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render Page correctly 1`] = `[Function]`;
